internal/host/validator.go

```diff
@@ -1626,3 +1634,38 @@ func (v *validator) noSkipMissingDisk(c *validationContext) (ValidationStatus, s
 	}
 	return ValidationSuccess, successMessage
 }
+
+
+
+func (v *validator) serviceCanConnectToSpokeKubeAPI(c *validationContext) (ValidationStatus, string) {
+	// When assisted-service operates with kubeApiEnabled for a day2 cluster, it will automatically sign CSR's by making a connection from the assisted-service to the spoke kubeApi. 
+	// This is for the purpose of signing CSR's and validating the completion of the install.
+	// This validation is to ensure that the assisted-service is able to connect to the spoke and has all required permissions so that we may detect any potential issues before installation proceeds.
+	// We therefore this validation will only cover this scenario and will skip otherwise.
+	if c.cluster == nil || !c.kubeApiEnabled || *c.cluster.Status != models.ClusterStatusAddingHosts {
+		return ValidationSuccessSuppressOutput, "Host is not part of a Day2 cluster, no need to check KubeAPI connectivity"
+	}
+
+	spokeK8sClientFactory := spoke_k8s_client.NewSpokeK8sClientFactory(v.log)
+	client, err := spokeK8sClientFactory.CreateFromSpokeKubeconfig(context.TODO(), c.host.ClusterID, c.objectHandler)
+	if err != nil {
+		return ValidationFailure, "Could not create the spoke k8s client connection using kubeconfig"
+	}
+
+	for _,check := range [][]string {
+		{"list", "CertificateSigningRequest"},
+		{"approve", "CertificateSigningRequest"},
+		{"list", "Node"},
+	} {
+		state, err := client.IsActionPermitted(check[0], check[1])
+		if err != nil {
+			return ValidationError, err.Error()
+		}
+		if !state {
+			return ValidationFailure, fmt.Sprintf("the spoke cluster credentials provided to assisted-service do not have permissions to %s %s on the spoke cluster", check[0], check[1])
```

Comment:
The credentials are (as of today) not "provided" to the assisted-service and the user has no way to change them. Perhaps "held by" is a better fit than "provided to"

Reply:
`to %s %s` sounds odd when the latter %s is singular, so perhaps change the loop iteratee to the plural (I don't think SAR would mind, but I'm not entirely sure)

Reply:
You can just chuck an extra s into the message rather than do that

`return ValidationFailure, fmt.Sprintf("the spoke cluster credentials provided to assisted-service do not have permissions to %s %ss on the spoke cluster", check[0], check[1]) `

Because all of the entities in the list are perfectly valid to express as plural in this messaging. No need to change what we send to kubernetes.